app/models/bishop.rb

```diff
@@ -4,4 +4,22 @@ def move_legal?(to_row, to_col)
     col_diff = (col - to_col).abs
     row_diff == col_diff
   end
+
+  def path_to_king
+    king = game.pieces.where.not(is_black: is_black).where(type: 'King').first
+    return nil unless (king.row - row).abs == (king.col - col).abs
```

Comment:
No problems when I change it to use `valid_move?`. However, I have to hold off on that for Queen since `Queen#move_legal?` doesn't exist in the master branch yet.